tc_pdf/handlers/pdf.py

```diff
@@ -63,21 +61,25 @@ def get(self, **kwargs):
 
         # Check if preview image already exists
         path = quote(preview_path.encode('utf-8'))
-        exists = yield gen.maybe_future(pdf.get(path))
+        exists = await pdf.get(path)
         if not exists:
             # create a new preview
-            data = yield self.create_preview(pdf_url)
+            data = await self.create_preview(pdf_url)
             if not data:
                 raise tornado.web.HTTPError(400)
             # store it in storage
-            yield gen.maybe_future(pdf.put(path, data))
+            await pdf.put(path, data)
         else:
             logger.debug('PDF preview already exists..')
 
         crypto = CryptoURL(key=self.context.server.security_key)
         options = {k: v for k, v in kwargs.items() if v and k != 'hash'}
         preview_url = crypto.generate(image_url=preview_path, **options)
-        kwargs['hash'] = RequestParser.path_to_parameters(preview_url)['hash']
+        parameters = RequestParser.path_to_parameters(preview_url)
+        for parameter in parameters:
+            if parameter[0] == 'hash':
```

Comment:
```suggestion
            if parameter[0] == 'hash' and parameter[1]:
```

Otherwise if unsafe URLs are allowed and an unsafe URL is provided there is an error:
```
2023-05-15 19:10:41 thumbor:ERROR ERROR: Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/tornado/web.py", line 1786, in _execute
    result = await result
  File "/app/tc_pdf/handlers/pdf.py", line 82, in get
    kwargs['hash'] = parameter[1].decode()
AttributeError: 'NoneType' object has no attribute 'decode'
```